### PR TITLE
enable mtls for control-api

### PR DESCRIPTION
**Pull request by kaitmore**
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

Enable TLS for gm-control-api
<br/>
<br/>
2. What **changes to custom.yaml** are required?

<br/>
<br/>

3. Have you documented any additional setup steps or configurations options?

<br/><br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

To test, assuming you have minikube running:

- update requirements.yaml to point to the local charts for control and gm-control-api
- install greymatter:

 ```
helm dep up greymatter
helm install greymatter -f greymatter.yaml -f greymatter-secrets.yaml -f credentials.yaml --set global.environment=kubernetes -n gm-deploy
```
- look at logs for gm-control-api and see that tls was enabled
- check you can reach the gm-control-api through the edge
- delete the deployment
- in greymatter.yaml, set global.control_api_tls=false and global.services.controlApi.egressTLS=false
- repeat previous steps with tls false to make sure everything is working
<br/>

### CODE REVIEW

**Comment by kaitmore** [comment]
@dgoldstein1 Thanks! You're right, charts should be bumped.

I took the helm install command from the Makefile. That was easiest for me since I already generated the `credentials.yaml` file by running `make credentials`

**Comment by tilleryd** [comment]
I also run the Makefile style install because it lessens copying or changing greymatter.yaml or greymatter-secrets.yaml. Having only credentials.yaml and the flag `--set global.environment=kubernetes` is all we need.

File: control/values.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ global:
   environment: kubernetes
   domain:
   route_url_name:
+  control_api_tls: true
```
Comment (tilleryd) [inline control/values.yaml]: This isn't a global config to me and seems more appropriate under `gm-control-api`.

Comment (tilleryd) [reply control/values.yaml]: I'm aware that you will need to change a bunch of files to address my concern, so I'm open to a debate.

File: gm-control-api/templates/gm-control-api.yaml [hunk 2]
```diff
@@ -75,6 +90,29 @@ spec:
...
+          # We give a full minute before we start health-checking for failures
+          livenessProbe:
+            exec:
+              command:
+                - /bin/sh
+                - -c
+                - "curl -k --cacert /etc/proxy/tls/sidecar/ca.crt --key /etc/proxy/tls/sidecar/server.key --cert /etc/proxy/tls/sidecar/server.crt https://localhost:5555/v1.0/cluster"
+            initialDelaySeconds: 60
+            periodSeconds: 3
+          {{- end }}
```
Comment (tilleryd) [inline gm-control-api/templates/gm-control-api.yaml]: Nice addition

**Comment by tilleryd** [comment]
I tested TLS and non-TLS, and it worked as intended. My only comment is on the global value.

File: gm-control-api/templates/gm-control-api.yaml (hunk 2 above)
Comment (chrisbsmith) [reply gm-control-api/templates/gm-control-api.yaml]: So I admittedly don't know enough about `livenessProbes` and `readinessProbes`, but was just reading this article today (https://learnk8s.io/production-best-practices/) and it recommends not using `livenessProbes` if possible.  And recommends the two probes be different.

This is really more food for thought, but we should definitely start including readiness probes in all our containers, and making sure they fail properly so k8s can restart them.

File: control/values.yaml (hunk 1 above)
Comment (kaitmore) [reply control/values.yaml]: @tilleryd This was the solution I discussed with @chrisbsmith. I agree it's not ideal, but the issue is that both control and control-api need to read this variable, and control wouldn't be able to if I kept it under gm-control-api as you suggested.

Open to other suggestions, it seemed messy to me to have to enable control-api tls in two places.

Comment (tilleryd) [reply control/values.yaml]: Very well, I defer to you guys on this one.

**Pull request merged**